src/coreclr/interpreter/compiler.cpp

```diff
@@ -2506,17 +2506,14 @@ void InterpCompiler::EmitCall(CORINFO_RESOLVED_TOKEN* constrainedClass, bool rea
                 }
                 m_pLastNewIns->data[0] = GetDataItemIndex(callInfo.hMethod);
             }
-            else if ((callInfo.classFlags & CORINFO_FLG_ARRAY) && !readonly)
+            else if ((callInfo.classFlags & CORINFO_FLG_ARRAY) && newObj)
             {
-                CORINFO_SIG_INFO ctorSignature;
-                CORINFO_CLASS_HANDLE ctorClass;
-
-                m_compHnd->getMethodSig(resolvedCallToken.hMethod, &ctorSignature);
-                ctorClass = m_compHnd->getMethodClass(resolvedCallToken.hMethod);
+                CORINFO_CLASS_HANDLE arrayClsHnd = m_compHnd->getMethodClass(resolvedCallToken.hMethod);
+                uint32_t rank = m_compHnd->getArrayRank(arrayClsHnd);
 
                 AddIns(INTOP_NEWMDARR);
-                m_pLastNewIns->data[0] = GetDataItemIndex(ctorClass);
-                m_pLastNewIns->data[1] = numArgs;
+                m_pLastNewIns->data[0] = GetDataItemIndex(arrayClsHnd);
+                m_pLastNewIns->data[1] = rank;
```

Comment:
This should not be rank. This should be the actual number of the arguments in the signature, so that the code handles different overloads of array constructor correctly. The earlier code looks correct, the new code looks wrong. I think this part of the change should be reverted.

Reply:
Also, you can rename the argument name on the executor side to numArgs to avoid confusion

https://github.com/dotnet/runtime/blob/ca63ce61949612a605a16ee88759947aecfb2e3e/src/coreclr/vm/interpexec.cpp#L135